sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/channel/SessionTracker.java

```diff
@@ -162,20 +121,8 @@ private void sendStartSessionIfNeeded() {
             /* New session: generate a new identifier. */
             mSid = UUIDUtils.randomUUID();
 
-            /* Update session map. */
-            mSessions.put(System.currentTimeMillis(), mSid);
-
-            /* Remove oldest session if we reached maximum storage capacity. */
-            if (mSessions.size() > STORAGE_MAX_SESSIONS) {
-                mSessions.pollFirstEntry();
-            }
-
-            /* Persist sessions. */
-            Set<String> sessionStorage = new HashSet<>();
-            for (Map.Entry<Long, UUID> session : mSessions.entrySet()) {
-                sessionStorage.add(session.getKey() + STORAGE_KEY_VALUE_SEPARATOR + session.getValue());
-            }
-            StorageHelper.PreferencesStorage.putStringSet(STORAGE_KEY, sessionStorage);
+            /* Update session storage. */
+            SessionContext.getInstance().addSession(mSid);
```

Comment:
Might this be done by `sendStartSessionIfNeeded`?

Reply:
that code is already inside sendStartSessionIfNeeded, not sure what you meant